addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -224,15 +224,19 @@ def action_clear_quantities_to_zero(self):
         self.picking_ids.filtered("show_clear_qty_button").action_clear_quantities_to_zero()
 
     def action_done(self):
+        def has_no_qty_done(picking):
+            return all(float_is_zero(line.qty_done, precision_rounding=line.product_uom_id.rounding) for line in picking.move_line_ids if line.state not in ('done', 'cancel'))
+
         self.ensure_one()
         self._check_company()
+        # Empty 'waiting for another operation' pickings will be removed from the batch when it is validated.
         pickings = self.mapped('picking_ids').filtered(lambda picking: picking.state not in ('cancel', 'done'))
-        if any(picking.state not in ('assigned', 'confirmed') for picking in pickings):
-            raise UserError(_('Some transfers are still waiting for goods. Please check or force their availability before setting this batch to done.'))
```

Comment:
probably need to delete this from the pot file

Reply:
Always forget about the .pot files. Thanks for reminding me!